src/type-layout.md

```diff
@@ -274,28 +295,42 @@ For all other enumerations, the layout is unspecified.
 
 Likewise, combining two primitive representations together is unspecified.
 
-### The `align` Representation
+### The `align` modifier
 
-The `align` representation can be used on `struct`s and `union`s to raise the
+The `align` modifier can be used on `struct`s and `union`s to raise the
 alignment of the type to a given value.
 
-Alignment is specified as a parameter in the form of `#[repr(align(x))]`. The
-alignment value must be a power of two of type `u32`. The `align` representation
-can raise the alignment of a type to be greater than it's primitive alignment,
-it cannot lower the alignment of a type.
+The alignment is specified as an integer parameter in the form of
+`#[repr(align(x))]`. The alignment value must be a power of two from 1 up to
+2<sup>29</sup>.
+
+The `align` modifier raises the type's alignment to the specified alignment.
+If the specified alignment is less than the alignment of the type without the
+`align` modifier, then the alignment is unaffected.
 
-The `align` and `packed` representations cannot be applied on the same type and
-a `packed` type cannot transitively contain another `align`ed type.
+The `align` and `packed` modifiers cannot be applied on the same type and a
+`packed` type cannot transitively contain another `align`ed type. `align` may
+only be applied to the default and `C` representations.
 
-### The `packed` Representation
+### The `packed` modifier
+
+The `packed` modifier can be used on `struct`s and `union`s to lower the
+alignment of the type to a given value.
 
-The `packed` representation can only be used on `struct`s and `union`s.
+The packing value is specified as an integer parameter in the form of
+`#[repr(packed(x))]`. If no value is given, as in `#[repr(packed)]`, then the
+packing value is 1. The packing value must be a power of two from 1 up to
+2<sup>29</sup>.
```

Comment:
If you extract out the wording about the default it is the same as the paragraph for `align`. So extract these two out into a common notion `repr_value` or something and then reference that one.